From 00000000000 Mon Sep 17 00:00:00 2001
From: Armbian Build System <build@armbian.com>
Date: Mon, 1 Jan 2024 00:00:00 +0000
Subject: [PATCH] rockchip: rv1106: Add Luckfox Pico Ultra W board support

This patch adds support for the Luckfox Pico Ultra W board which uses
the Rockchip RV1106 SoC. The board supports both SPI NAND and eMMC
storage options.

Signed-off-by: Armbian Build System <build@armbian.com>
---
 configs/luckfox_rv1106_spi_nand_tb_defconfig | 123 ++++++++++++++++++++++++++
 configs/luckfox_rv1106_uboot_defconfig        | 144 ++++++++++++++++++++++++++++
 configs/luckfox_rv1106_uboot_emmc_tb_defconfig| 120 ++++++++++++++++++++++++++
 arch/arm/dts/rv1106-luckfox-emmc-tb.dts      | 130 ++++++++++++++++++++++++++
 arch/arm/dts/rv1106-luckfox-spi-nand-tb.dts  | 132 ++++++++++++++++++++++++++
 arch/arm/dts/rv1106-luckfox.dts               |  47 ++++++++++
 6 files changed, 696 insertions(+)
 create mode 100644 configs/luckfox_rv1106_spi_nand_tb_defconfig
 create mode 100644 configs/luckfox_rv1106_uboot_defconfig
 create mode 100644 configs/luckfox_rv1106_uboot_emmc_tb_defconfig
 create mode 100644 arch/arm/dts/rv1106-luckfox-emmc-tb.dts
 create mode 100644 arch/arm/dts/rv1106-luckfox-spi-nand-tb.dts
 create mode 100644 arch/arm/dts/rv1106-luckfox.dts

diff --git a/configs/luckfox_rv1106_spi_nand_tb_defconfig b/configs/luckfox_rv1106_spi_nand_tb_defconfig
new file mode 100644
index 00000000000..dd342c1a008
--- /dev/null
+++ b/configs/luckfox_rv1106_spi_nand_tb_defconfig
@@ -0,0 +1,124 @@
+CONFIG_ARM=y
+CONFIG_TPL=y
+CONFIG_TPL_NEEDS_SEPARATE_TEXT_BASE=y
+CONFIG_TPL_TEXT_BASE=0x0
+CONFIG_ARCH_ROCKCHIP=y
+CONFIG_SPL_LIBCOMMON_SUPPORT=y
+CONFIG_SPL_LIBGENERIC_SUPPORT=y
+CONFIG_SYS_MALLOC_F_LEN=0x80000
+CONFIG_SPL_FIT_GENERATOR="arch/arm/mach-rockchip/make_fit_optee.sh"
+CONFIG_ROCKCHIP_RV1106=y
+CONFIG_ROCKCHIP_SPL_RESERVE_IRAM=0x0
+CONFIG_ROCKCHIP_FIT_IMAGE=y
+CONFIG_USING_KERNEL_DTB_V2=y
+CONFIG_ROCKCHIP_META=y
+CONFIG_ROCKCHIP_FIT_IMAGE_PACK=y
+# CONFIG_ROCKCHIP_SET_SN is not set
+# CONFIG_ROCKCHIP_SET_ETHADDR is not set
+CONFIG_LOADER_INI="RV1106MINIALL_SPI_NAND_TB.ini"
+CONFIG_TRUST_INI="RV1106TOS_TB.ini"
+CONFIG_SPL_SERIAL_SUPPORT=y
+CONFIG_SPL_DRIVERS_MISC_SUPPORT=y
+CONFIG_TARGET_EVB_RV1106=y
+CONFIG_SPL_LIBDISK_SUPPORT=y
+CONFIG_SPL_SPI_FLASH_SUPPORT=y
+CONFIG_SPL_SPI_SUPPORT=y
+CONFIG_DEFAULT_DEVICE_TREE="rv1106-luckfox-spi-nand-tb"
+CONFIG_DEBUG_UART=y
+# CONFIG_DISTRO_DEFAULTS is not set
+CONFIG_FIT=y
+CONFIG_FIT_IMAGE_POST_PROCESS=y
+CONFIG_SPL_LOAD_FIT=y
+CONFIG_SPL_FIT_IMAGE_POST_PROCESS=y
+CONFIG_SPL_FIT_HW_CRYPTO=y
+# CONFIG_SPL_SYS_DCACHE_OFF is not set
+CONFIG_SPL_FIT_IMAGE_KB=512
+CONFIG_SPL_FIT_IMAGE_MULTIPLE=1
+CONFIG_BOOTDELAY=0
+CONFIG_SYS_CONSOLE_INFO_QUIET=y
+# CONFIG_DISPLAY_CPUINFO is not set
+# CONFIG_SKIP_RELOCATE_UBOOT is not set
+CONFIG_SPL_ADC_SUPPORT=y
+CONFIG_SPL_BOARD_INIT=y
+# CONFIG_SPL_RAW_IMAGE_SUPPORT is not set
+# CONFIG_SPL_LEGACY_IMAGE_SUPPORT is not set
+CONFIG_SPL_MTD_SUPPORT=y
+CONFIG_SPL_KERNEL_BOOT=y
+CONFIG_HUSH_PARSER=y
+# CONFIG_CMD_BDI is not set
+# CONFIG_CMD_CONSOLE is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_IMI is not set
+# CONFIG_CMD_IMLS is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_CRC32 is not set
+# CONFIG_CMD_FLASH is not set
+# CONFIG_CMD_FPGA is not set
+# CONFIG_CMD_LOADB is not set
+# CONFIG_CMD_LOADS is not set
+# CONFIG_CMD_ITEST is not set
+# CONFIG_CMD_SOURCE is not set
+# CONFIG_CMD_SETEXPR is not set
+# CONFIG_CMD_NET is not set
+# CONFIG_CMD_NFS is not set
+# CONFIG_CMD_MISC is not set
+CONFIG_CMD_MTD_BLK=y
+CONFIG_SPL_OF_CONTROL=y
+CONFIG_SPL_DTB_MINIMUM=y
+CONFIG_OF_LIVE=y
+CONFIG_OF_SPL_REMOVE_PROPS="pinctrl-0 pinctrl-names interrupt-parent assigned-clocks assigned-clock-rates assigned-clock-parents"
+CONFIG_OF_U_BOOT_REMOVE_PROPS="interrupt-parent"
+CONFIG_ENVF=y
+CONFIG_REGMAP=y
+CONFIG_SPL_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_SPL_SYSCON=y
+# CONFIG_SARADC_ROCKCHIP is not set
+CONFIG_SARADC_ROCKCHIP_V2=y
+CONFIG_SPL_BLK_READ_PREPARE=y
+CONFIG_CLK=y
+CONFIG_SPL_CLK=y
+CONFIG_SPL_DM_CRYPTO=y
+CONFIG_SPL_ROCKCHIP_CRYPTO_V2=y
+CONFIG_ROCKCHIP_GPIO=y
+# CONFIG_DM_I2C is not set
+CONFIG_SPL_INPUT=y
+CONFIG_DM_KEY=y
+CONFIG_ADC_KEY=y
+CONFIG_SPL_ADC_KEY=y
+CONFIG_SPL_MISC=y
+CONFIG_SPL_MISC_DECOMPRESS=y
+CONFIG_SPL_ROCKCHIP_HW_DECOMPRESS=y
+# CONFIG_MMC is not set
+# CONFIG_DM_MMC is not set
+CONFIG_MTD=y
+CONFIG_MTD_BLK=y
+CONFIG_MTD_DEVICE=y
+CONFIG_MTD_SPI_NAND=y
+CONFIG_SPI_FLASH=y
+CONFIG_SF_DEFAULT_MODE=0x1
+CONFIG_SF_DEFAULT_SPEED=50000000
+CONFIG_SPI_FLASH_GIGADEVICE=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_SPI_FLASH_XMC=y
+CONFIG_SPI_FLASH_MTD=y
+CONFIG_PINCTRL=y
+# CONFIG_DM_REGULATOR is not set
+# CONFIG_DM_PWM is not set
+CONFIG_RAM=y
+CONFIG_TPL_RAM=y
+CONFIG_ROCKCHIP_SDRAM_COMMON=y
+CONFIG_DM_RESET=y
+CONFIG_SPL_DM_RESET=y
+CONFIG_SPL_RESET_ROCKCHIP=y
+CONFIG_BAUDRATE=1500000
+CONFIG_DEBUG_UART_BASE=0xff4c0000
+CONFIG_DEBUG_UART_CLOCK=24000000
+CONFIG_DEBUG_UART_SHIFT=2
+CONFIG_ROCKCHIP_SFC=y
+CONFIG_SYSRESET=y
+# CONFIG_SYSRESET_SYSCON_REBOOT is not set
+CONFIG_USE_TINY_PRINTF=y
+CONFIG_SPL_TINY_MEMSET=y
+CONFIG_ERRNO_STR=y
+# CONFIG_EFI_LOADER is not set


diff --git a/configs/luckfox_rv1106_uboot_defconfig b/configs/luckfox_rv1106_uboot_defconfig
new file mode 100644
index 00000000000..a8a150daba4
--- /dev/null
+++ b/configs/luckfox_rv1106_uboot_defconfig
@@ -0,0 +1,143 @@
+CONFIG_ARM=y
+CONFIG_TPL=y
+CONFIG_TPL_NEEDS_SEPARATE_TEXT_BASE=y
+CONFIG_TPL_TEXT_BASE=0x0
+CONFIG_ARCH_ROCKCHIP=y
+CONFIG_SPL_GPIO_SUPPORT=y
+CONFIG_SPL_LIBCOMMON_SUPPORT=y
+CONFIG_SPL_LIBGENERIC_SUPPORT=y
+CONFIG_SYS_MALLOC_F_LEN=0x80000
+CONFIG_SPL_FIT_GENERATOR="arch/arm/mach-rockchip/make_fit_optee.sh"
+CONFIG_ROCKCHIP_RV1106=y
+CONFIG_ROCKCHIP_SPL_RESERVE_IRAM=0x0
+CONFIG_ROCKCHIP_FIT_IMAGE=y
+CONFIG_USING_KERNEL_DTB_V2=y
+CONFIG_ROCKCHIP_FIT_IMAGE_PACK=y
+CONFIG_ROCKCHIP_CMD="sd_update -"
+CONFIG_SPL_SERIAL_SUPPORT=y
+CONFIG_SPL_DRIVERS_MISC_SUPPORT=y
+CONFIG_TARGET_EVB_RV1106=y
+CONFIG_SPL_LIBDISK_SUPPORT=y
+CONFIG_SPL_SPI_FLASH_SUPPORT=y
+CONFIG_SPL_SPI_SUPPORT=y
+CONFIG_DEFAULT_DEVICE_TREE="rv1106-luckfox"
+CONFIG_DEBUG_UART=y
+# CONFIG_DISTRO_DEFAULTS is not set
+CONFIG_ANDROID_BOOT_IMAGE=y
+CONFIG_FIT=y
+CONFIG_FIT_HW_CRYPTO=y
+CONFIG_SPL_LOAD_FIT=y
+CONFIG_SPL_FIT_IMAGE_POST_PROCESS=y
+CONFIG_SPL_FIT_HW_CRYPTO=y
+# CONFIG_SPL_SYS_DCACHE_OFF is not set
+CONFIG_SPL_FIT_IMAGE_KB=256
+CONFIG_SPL_FIT_IMAGE_MULTIPLE=1
+CONFIG_BOOTDELAY=0
+CONFIG_SYS_CONSOLE_INFO_QUIET=y
+# CONFIG_DISPLAY_CPUINFO is not set
+CONFIG_ANDROID_BOOTLOADER=y
+CONFIG_ANDROID_BOOT_IMAGE_HASH=y
+# CONFIG_SKIP_RELOCATE_UBOOT is not set
+CONFIG_SPL_BOARD_INIT=y
+# CONFIG_SPL_RAW_IMAGE_SUPPORT is not set
+# CONFIG_SPL_LEGACY_IMAGE_SUPPORT is not set
+CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_PARTITION=y
+CONFIG_SPL_MTD_SUPPORT=y
+CONFIG_SPL_AB=y
+CONFIG_HUSH_PARSER=y
+# CONFIG_CMD_BDI is not set
+# CONFIG_CMD_CONSOLE is not set
+CONFIG_CMD_BOOTZ=y
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_IMI is not set
+# CONFIG_CMD_IMLS is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_CRC32 is not set
+# CONFIG_CMD_FLASH is not set
+# CONFIG_CMD_FPGA is not set
+CONFIG_CMD_GPIO=y
+CONFIG_RANDOM_UUID=y
+# CONFIG_CMD_LOADB is not set
+# CONFIG_CMD_LOADS is not set
+# CONFIG_CMD_BOOT_ANDROID=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_MTD=y
+CONFIG_CMD_PART=y
+# CONFIG_CMD_ITEST is not set
+CONFIG_CMD_SCRIPT_UPDATE=y
+# CONFIG_CMD_SOURCE is not set
+# CONFIG_CMD_SETEXPR is not set
+CONFIG_CMD_TFTP_BOOTM=y
+CONFIG_CMD_TFTP_FLASH=y
+CONFIG_CMD_DHCP=y
+# CONFIG_CMD_NFS is not set
+CONFIG_CMD_PING=y
+# CONFIG_CMD_MISC is not set
+CONFIG_CMD_FAT=y
+# CONFIG_SPL_DOS_PARTITION is not set
+CONFIG_SPL_OF_CONTROL=y
+CONFIG_SPL_DTB_MINIMUM=y
+CONFIG_OF_LIVE=y
+CONFIG_OF_SPL_REMOVE_PROPS="interrupt-parent assigned-clocks assigned-clock-rates assigned-clock-parents"
+CONFIG_OF_U_BOOT_REMOVE_PROPS="interrupt-parent"
+CONFIG_ENVF=y
+CONFIG_ENVF_LIST="blkdevparts mtdparts sys_bootargs app reserved ipaddr serverip netmask gatewayip ethaddr"
+CONFIG_REGMAP=y
+CONFIG_SPL_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_SPL_SYSCON=y
+# CONFIG_SARADC_ROCKCHIP is not set
+CONFIG_SARADC_ROCKCHIP_V2=y
+CONFIG_CLK=y
+CONFIG_SPL_CLK=y
+CONFIG_DM_CRYPTO=y
+CONFIG_SPL_DM_CRYPTO=y
+CONFIG_ROCKCHIP_CRYPTO_V2=y
+CONFIG_SPL_ROCKCHIP_CRYPTO_V2=y
+CONFIG_DM_RNG=y
+CONFIG_RNG_ROCKCHIP=y
+CONFIG_ROCKCHIP_GPIO=y
+# CONFIG_DM_I2C is not set
+CONFIG_DM_KEY=y
+CONFIG_ADC_KEY=y
+CONFIG_MISC=y
+CONFIG_SPL_MISC=y
+CONFIG_ROCKCHIP_OTP=y
+CONFIG_SPL_ROCKCHIP_SECURE_OTP=y
+# CONFIG_SUPPORT_EMMC_RPMB is not set
+CONFIG_MMC_DW=y
+CONFIG_MMC_DW_ROCKCHIP=y
+CONFIG_MTD=y
+CONFIG_MTD_BLK=y
+CONFIG_MTD_DEVICE=y
+CONFIG_MTD_SPI_NAND=y
+CONFIG_PHY_RK630=y
+CONFIG_DM_ETH=y
+CONFIG_DM_ETH_PHY=y
+CONFIG_DWC_ETH_QOS=y
+# CONFIG_DWC_ETH_QOS_FULL is not set
+CONFIG_GMAC_ROCKCHIP=y
+CONFIG_PINCTRL=y
+CONFIG_SPL_PINCTRL=y
+# CONFIG_DM_REGULATOR is not set
+# CONFIG_DM_PWM is not set
+CONFIG_RAM=y
+CONFIG_TPL_RAM=y
+CONFIG_ROCKCHIP_SDRAM_COMMON=y
+CONFIG_DM_RESET=y
+CONFIG_DEBUG_UART_BASE=0xff4c0000
+CONFIG_ROCKCHIP_SFC=y
+CONFIG_SYSRESET=y
+# CONFIG_SYSRESET_SYSCON_REBOOT is not set
+# CONFIG_FAT_WRITE is not set
+CONFIG_USE_TINY_PRINTF=y
+# CONFIG_REGEX is not set
+CONFIG_SPL_TINY_MEMSET=y
+CONFIG_RSA=y
+CONFIG_SPL_RSA=y
+CONFIG_RSA_N_SIZE=0x200
+CONFIG_RSA_E_SIZE=0x10
+CONFIG_RSA_C_SIZE=0x20
+CONFIG_SPL_LZMA=y
+CONFIG_SPL_GZIP=y
+CONFIG_ERRNO_STR=y
+# CONFIG_EFI_LOADER is not set


diff --git a/configs/luckfox_rv1106_uboot_emmc_tb_defconfig b/configs/luckfox_rv1106_uboot_emmc_tb_defconfig
new file mode 100644
index 00000000000..c92d0393a29
--- /dev/null
+++ b/configs/luckfox_rv1106_uboot_emmc_tb_defconfig
@@ -0,0 +1,117 @@
+CONFIG_ARM=y
+CONFIG_TPL=y
+CONFIG_TPL_NEEDS_SEPARATE_TEXT_BASE=y
+CONFIG_TPL_TEXT_BASE=0x0
+CONFIG_ARCH_ROCKCHIP=y
+CONFIG_SPL_LIBCOMMON_SUPPORT=y
+CONFIG_SPL_LIBGENERIC_SUPPORT=y
+CONFIG_SYS_MALLOC_F_LEN=0x100000
+CONFIG_SPL_FIT_GENERATOR="arch/arm/mach-rockchip/make_fit_optee.sh"
+CONFIG_ROCKCHIP_RV1106=y
+CONFIG_ROCKCHIP_SPL_RESERVE_IRAM=0x0
+CONFIG_ROCKCHIP_FIT_IMAGE=y
+CONFIG_USING_KERNEL_DTB_V2=y
+CONFIG_ROCKCHIP_META=y
+CONFIG_ROCKCHIP_FIT_IMAGE_PACK=y
+# CONFIG_ROCKCHIP_SET_SN is not set
+# CONFIG_ROCKCHIP_SET_ETHADDR is not set
+CONFIG_LOADER_INI="RV1106MINIALL_EMMC_TB.ini"
+CONFIG_TRUST_INI="RV1106TOS_TB.ini"
+CONFIG_SPL_SERIAL_SUPPORT=y
+CONFIG_SPL_DRIVERS_MISC_SUPPORT=y
+CONFIG_TARGET_EVB_RV1106=y
+CONFIG_SPL_LIBDISK_SUPPORT=y
+CONFIG_DEFAULT_DEVICE_TREE="rv1106-luckfox-emmc-tb"
+CONFIG_DEBUG_UART=y
+# CONFIG_DISTRO_DEFAULTS is not set
+CONFIG_SPL_SYS_MALLOC_F_LEN=0x80000
+CONFIG_TPL_SYS_MALLOC_F_LEN=0x80000
+CONFIG_ANDROID_BOOT_IMAGE=y
+CONFIG_FIT=y
+CONFIG_FIT_IMAGE_POST_PROCESS=y
+CONFIG_SPL_LOAD_FIT=y
+CONFIG_SPL_FIT_IMAGE_POST_PROCESS=y
+CONFIG_SPL_FIT_HW_CRYPTO=y
+# CONFIG_SPL_SYS_DCACHE_OFF is not set
+CONFIG_SPL_FIT_IMAGE_KB=256
+CONFIG_SPL_FIT_IMAGE_MULTIPLE=1
+CONFIG_BOOTDELAY=0
+CONFIG_SYS_CONSOLE_INFO_QUIET=y
+# CONFIG_DISPLAY_CPUINFO is not set
+CONFIG_ANDROID_BOOTLOADER=y
+CONFIG_ANDROID_BOOT_IMAGE_HASH=y
+# CONFIG_SKIP_RELOCATE_UBOOT is not set
+CONFIG_SPL_ADC_SUPPORT=y
+CONFIG_SPL_BOARD_INIT=y
+# CONFIG_SPL_RAW_IMAGE_SUPPORT is not set
+# CONFIG_SPL_LEGACY_IMAGE_SUPPORT is not set
+CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_PARTITION=y
+CONFIG_SPL_MMC_WRITE=y
+CONFIG_SPL_KERNEL_BOOT=y
+CONFIG_HUSH_PARSER=y
+# CONFIG_CMD_BDI is not set
+# CONFIG_CMD_CONSOLE is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_IMI is not set
+# CONFIG_CMD_IMLS is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_CRC32 is not set
+# CONFIG_CMD_FLASH is not set
+# CONFIG_CMD_FPGA is not set
+# CONFIG_CMD_LOADB is not set
+# CONFIG_CMD_LOADS is not set
+# CONFIG_CMD_ITEST is not set
+# CONFIG_CMD_SOURCE is not set
+# CONFIG_CMD_SETEXPR is not set
+# CONFIG_CMD_NET is not set
+# CONFIG_CMD_NFS is not set
+# CONFIG_CMD_MISC is not set
+CONFIG_SPL_OF_CONTROL=y
+CONFIG_SPL_DTB_MINIMUM=y
+CONFIG_OF_LIVE=y
+CONFIG_OF_SPL_REMOVE_PROPS="pinctrl-0 pinctrl-names interrupt-parent assigned-clocks assigned-clock-rates assigned-clock-parents"
+CONFIG_OF_U_BOOT_REMOVE_PROPS="interrupt-parent"
+CONFIG_ENVF=y
+CONFIG_REGMAP=y
+CONFIG_SPL_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_SPL_SYSCON=y
+# CONFIG_SARADC_ROCKCHIP is not set
+CONFIG_SARADC_ROCKCHIP_V2=y
+CONFIG_SPL_ADC_KEY=y
+CONFIG_SPL_BLK_READ_PREPARE=y
+CONFIG_CLK=y
+CONFIG_SPL_CLK=y
+CONFIG_SPL_DM_CRYPTO=y
+CONFIG_SPL_ROCKCHIP_CRYPTO_V2=y
+CONFIG_ROCKCHIP_GPIO=y
+# CONFIG_DM_I2C is not set
+CONFIG_SPL_INPUT=y
+CONFIG_DM_KEY=y
+CONFIG_ADC_KEY=y
+CONFIG_SPL_ADC_KEY=y
+CONFIG_SPL_MISC=y
+CONFIG_SPL_MISC_DECOMPRESS=y
+CONFIG_SPL_ROCKCHIP_HW_DECOMPRESS=y
+# CONFIG_SUPPORT_EMMC_RPMB is not set
+CONFIG_MMC_DW=y
+CONFIG_MMC_DW_ROCKCHIP=y
+CONFIG_MMC_USE_PRE_CONFIG=y
+CONFIG_SF_DEFAULT_MODE=0x3
+CONFIG_PINCTRL=y
+# CONFIG_DM_REGULATOR is not set
+# CONFIG_DM_PWM is not set
+CONFIG_RAM=y
+CONFIG_TPL_RAM=y
+CONFIG_ROCKCHIP_SDRAM_COMMON=y
+CONFIG_DM_RESET=y
+CONFIG_SPL_DM_RESET=y
+CONFIG_SPL_RESET_ROCKCHIP=y
+CONFIG_DEBUG_UART_BASE=0xff4c0000
+CONFIG_DEBUG_UART_CLOCK=24000000
+CONFIG_DEBUG_UART_SHIFT=2
+CONFIG_SYSRESET=y
+# CONFIG_SYSRESET_SYSCON_REBOOT is not set
+CONFIG_USE_TINY_PRINTF=y
+CONFIG_SPL_TINY_MEMSET=y
+CONFIG_ERRNO_STR=y
+# CONFIG_EFI_LOADER is not set


diff --git a/arch/arm/dts/rv1106-luckfox-emmc-tb.dts b/arch/arm/dts/rv1106-luckfox-emmc-tb.dts
new file mode 100644
index 00000000000..e7f8c9d1b2f
--- /dev/null
+++ b/arch/arm/dts/rv1106-luckfox-emmc-tb.dts
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2022 Rockchip Electronics Co., Ltd.
+ */
+
+/dts-v1/;
+
+#include "rv1106.dtsi"
+#include "rv1106-u-boot.dtsi"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Rockchip RV1106 EVB2 Board";
+	compatible = "rockchip,rv1106-evb2", "rockchip,rv1106";
+
+	chosen {
+		stdout-path = &uart2;
+		u-boot,spl-boot-order = &spi_nor, &emmc;
+	};
+
+	adc-keys {
+		compatible = "adc-keys";
+		io-channels = <&saradc 0>;
+		io-channel-names = "buttons";
+		poll-interval = <100>;
+		keyup-threshold-microvolt = <1800000>;
+		u-boot,dm-spl;
+		status = "okay";
+
+		volumeup-key {
+			u-boot,dm-spl;
+			linux,code = <KEY_VOLUMEUP>;
+			label = "volume up";
+			press-threshold-microvolt = <0>;
+		};
+
+		volumedown-key {
+			u-boot,dm-spl;
+			linux,code = <KEY_VOLUMEDOWN>;
+			label = "volume down";
+			press-threshold-microvolt = <400781>;
+		};
+	};
+
+};
+
+&hw_decompress {
+	u-boot,dm-spl;
+	status = "okay";
+};
+
+//&gmac {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&mdio {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&pinctrl {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&pcfg_pull_up_drv_level_2 {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&pcfg_pull_up {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&gpio3 {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&rmii_phy {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&rmii_phy {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+&saradc {
+	u-boot,dm-spl;
+	status = "okay";
+};
+
+&sdmmc {
+	/delete-property/ u-boot,dm-spl;
+	status = "disabled";
+};
+
+&sdmmc0 {
+	/delete-property/ u-boot,dm-spl;
+};
+
+&sdmmc0_bus4 {
+	/delete-property/ u-boot,dm-spl;
+};
+
+&sdmmc0_clk {
+	/delete-property/ u-boot,dm-spl;
+};
+
+&sdmmc0_cmd {
+	/delete-property/ u-boot,dm-spl;
+};
+
+&sdmmc0_det {
+	/delete-property/ u-boot,dm-spl;
+};
+
+&spi_nand {
+	/delete-property/ u-boot,dm-spl;
+	status = "disabled";
+};
+
+&spi_nor {
+	spi-max-frequency = <125000000>;
+};


diff --git a/arch/arm/dts/rv1106-luckfox-spi-nand-tb.dts b/arch/arm/dts/rv1106-luckfox-spi-nand-tb.dts
new file mode 100644
index 00000000000..f9a1b2c3d4e
--- /dev/null
+++ b/arch/arm/dts/rv1106-luckfox-spi-nand-tb.dts
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2022 Rockchip Electronics Co., Ltd.
+ */
+
+/dts-v1/;
+
+#include "rv1106.dtsi"
+#include "rv1106-u-boot.dtsi"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Rockchip RV1106 EVB2 Board";
+	compatible = "rockchip,rv1106-evb2", "rockchip,rv1106";
+
+	chosen {
+		stdout-path = &uart2;
+		u-boot,spl-boot-order = &spi_nand, &emmc;
+	};
+
+	adc-keys {
+		compatible = "adc-keys";
+		io-channels = <&saradc 0>;
+		io-channel-names = "buttons";
+		poll-interval = <100>;
+		keyup-threshold-microvolt = <1800000>;
+		u-boot,dm-spl;
+		status = "okay";
+
+		volumeup-key {
+			u-boot,dm-spl;
+			linux,code = <KEY_VOLUMEUP>;
+			label = "volume up";
+			press-threshold-microvolt = <0>;
+		};
+
+		volumedown-key {
+			u-boot,dm-spl;
+			linux,code = <KEY_VOLUMEDOWN>;
+			label = "volume down";
+			press-threshold-microvolt = <400781>;
+		};
+	};
+
+};
+
+&hw_decompress {
+	u-boot,dm-spl;
+	status = "okay";
+};
+
+//&gmac {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&mdio {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&pinctrl {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&pcfg_pull_up_drv_level_2 {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&pcfg_pull_up {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&gpio3 {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&rmii_phy {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+//&rng {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+&saradc {
+	u-boot,dm-spl;
+	status = "okay";
+};
+
+&sdmmc {
+	/delete-property/ u-boot,dm-spl;
+	status = "disabled";
+};
+
+&sdmmc0 {
+	/delete-property/ u-boot,dm-spl;
+};
+
+&sdmmc0_bus4 {
+	/delete-property/ u-boot,dm-spl;
+};
+
+&sdmmc0_clk {
+	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+};
+
+&sdmmc0_cmd {
+	/delete-property/ u-boot,dm-spl;
+};
+
+&sdmmc0_det {
+	/delete-property/ u-boot,dm-spl;
+};
+
+//&spi_nand {
+//	/delete-property/ u-boot,dm-spl;
+//	status = "disabled";
+//};
+
+&spi_nor {
+	/delete-property/ u-boot,dm-spl;
+	status = "disabled";
+	//spi-max-frequency = <125000000>;
+};


diff --git a/arch/arm/dts/rv1106-luckfox.dts b/arch/arm/dts/rv1106-luckfox.dts
new file mode 100644
index 00000000000..5a4b3c2d1e0
--- /dev/null
+++ b/arch/arm/dts/rv1106-luckfox.dts
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2022 Rockchip Electronics Co., Ltd.
+ */
+
+/dts-v1/;
+
+#include "rv1106.dtsi"
+#include "rv1106-u-boot.dtsi"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Rockchip RV1106 EVB2 Board";
+	compatible = "rockchip,rv1106-evb2", "rockchip,rv1106";
+
+	chosen {
+		stdout-path = &uart2;
+		u-boot,spl-boot-order = &sdmmc, &spi_nand, &emmc;
+	};
+
+	adc-keys {
+		compatible = "adc-keys";
+		io-channels = <&saradc 0>;
+		io-channel-names = "buttons";
+		keyup-threshold-microvolt = <1800000>;
+		u-boot,dm-spl;
+		status = "okay";
+
+		volumeup-key {
+			u-boot,dm-spl;
+			linux,code = <KEY_VOLUMEUP>;
+			label = "volume up";
+			press-threshold-microvolt = <1750>;
+		};
+
+	};
+};
+
+&hw_decompress {
+	u-boot,dm-spl;
+	status = "okay";
+};
+
+&saradc {
+	u-boot,dm-spl;
+	status = "okay";
+};
--